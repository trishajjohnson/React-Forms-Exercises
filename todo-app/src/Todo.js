import React from "react";

function Todo({ id, handleRemove, task }) {
    const remove = () => handleRemove(id);
    
    return (
        <div>
            <div>
                {task}
            </div>
            <button onClick={remove}>X</button>
        </div> 
    );
}
  
  export default Todo;
  