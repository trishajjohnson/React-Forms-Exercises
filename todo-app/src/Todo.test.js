import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo";

// Smoke test for Todo component
it("should render without crashing", function() {
  render(<Todo />)
});

// Snapshot test for Todo component
it("should match snapshot", function() {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});