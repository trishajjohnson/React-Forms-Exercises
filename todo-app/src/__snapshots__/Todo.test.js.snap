// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
